src/InfraestruturaDeDados/Repositorios/RepositorioSqlServer.cs

```diff
@@ -37,13 +38,26 @@ public int Criar(ProdutoTapecaria produtoTapecaria)
             return produtoTapecaria.Id;
         }
 
-        public List<ProdutoTapecaria> ObterTodos()
+        public List<ProdutoTapecaria> ObterTodos(string? tipo, string? detalhes)
         {
             var conexaoSql = new SqlConnection(_connectionString);
 
             var query = @"SELECT * 
-                        FROM tb_Tapecaria 
-                        ORDER BY Id";
+                        FROM tb_Tapecaria ";
+
+            if (tipo is not null) 
```

Comment:
Acho que ainda dá pra otimizar isso aqui, pra ao menos evitar a repetição do código do like

Reply:
Estou tento dificuldades em enxergar uma forma menos extensa. Pensei no seguinte:

if (tipo is not null || detalhes is not null) 
{
    query += $"WHERE ";

    if (tipo is not null)
    {
        query += $"Tipo = {tipo} ";
    }

    if (tipo is not null & detalhes is not null)
    {
        query += $"AND ";
    }

    if (detalhes is not null)
    {
        query += $"LOWER(Detalhes) LIKE LOWER('%{detalhes}%') ";
    }
}

Tem uma sugestão diferente?